pkgs/development/tools/misc/remarkable/remarkable-toolchain/default.nix

```diff
@@ -0,0 +1,36 @@
+{ stdenv, fetchurl, libarchive, python3, file }:
+
+stdenv.mkDerivation rec {
+  pname = "remarkable-toolchain";
+  version = "1.8-23.9.2019";
+
+  src = fetchurl {
+    url = "https://remarkable.engineering/oecore-x86_64-cortexa9hf-neon-toolchain-zero-gravitas-${version}.sh";
+    sha256 = "6299955721bcd9bef92a87ad3cfe4d31df8e2da95b0c4b2cdded4431aa6748b0";
+  };
+
+  nativeBuildInputs = [
+    libarchive
+    python3
+    file
+  ];
+
+  unpackCmd = "mkdir src; install $curSrc src/install-toolchain.sh";
+
+  dontBuild = true;
+
+  installPhase = ''
+    patchShebangs install-toolchain.sh
+    sed -i -e '3,9d' install-toolchain.sh # breaks PATH
+    sed -i 's|PYTHON=.*$|PYTHON=${python3}/bin/python|' install-toolchain.sh
+    ./install-toolchain.sh -D -y -d $out
```

Comment:
Can you say a bit about what this is doing, e.g. in a comment here? Does this build from source or is it a binary distribution? It's a bit awkward for me to take a look myself, as unpacking a shar archive requires a lot of trust.

Also, could you add a comment somewhere in this file linking to the source in some browsable form? Immediate motivation is that I was trying to double-check the licence meta field and am sort of stuck, but I think it would be helpful more generally.

Reply:
Sure; the `.sh` file is a small shell script and then a compressed archive appended on to the end of it. The shell script is an installer, which can be read by opening it in a text editor rather than running it (text editor needs to support large files, and probably will chug a bit on loading). This toolchain is a binary distribution. The purpose of this package is essentially to run this installer, and then let `autoPatchelfHook` do its magic to ensure that the resulting binaries work. I did try pretty hard to just extract the embedded archive and then do the install in the nix expression, but it kept breaking and requiring new hacks to make progress - in the end, it turned out to be not that hard to get the original install script to work. 

Lines 3-9 are the following:
```
[ -z "$ENVCLEANED" ] && exec /usr/bin/env -i ENVCLEANED=1 HOME="$HOME" \
	LC_ALL=en_US.UTF-8 \
	TERM=$TERM \
	http_proxy="$http_proxy" https_proxy="$https_proxy" ftp_proxy="$ftp_proxy" \
	no_proxy="$no_proxy" all_proxy="$all_proxy" GIT_PROXY_COMMAND="$GIT_PROXY_COMMAND" "$0" "$@"
[ -f /etc/environment ] && . /etc/environment
export PATH=`echo "$PATH" | sed -e 's/:\.//' -e 's/::/:/'`
```

which messes up the path. Removing those lines fixes this issue.

I actually don't know where the source comes from; I think it is cobbled together from various repositories in https://github.com/reMarkable. I actually guessed that the license was GPLv2 because that's what Linux uses.

Reply:
Thanks for the details!

I doubt I know more than you about conventions for nixpkgs (I'm mostly just a user), so don't let me block you (you have power to merge, right?), but here are things to consider doing before merging:
* Please add a comment somewhere saying it's a binary distribution (kind of hard to tell what's going on here otherwise).
* Would it make sense to rename the expression to end in "-bin" or something, e.g. remarkable-toolchain-bin? I'm not sure if there's a convention for binary packages but I think I remember there was a `firefox-bin` at some point. My default expectation is that expressions are for building from source.
* I wonder if it would be safer to set the licence to `unfree` or `unfreeRedistributable` with a comment saying it's probably GPL?  I don't know what the risks are for associating the wrong licence.

Thanks for your contribution!

Reply:
Thanks for your review. I thought it was clear that it's not built from source from `dontBuild = true;` As far as I'm aware, there is no such `-bin` convention in nixpkgs (I've gotten binary packages merged that don't adhere to such a convention - I believe this is especially true when source is not available). One of the problems with setting the license to `unfree` is that it doesn't get built by hydra, so when a user tries to install the package they'll have to download the source tarball and then build it, using ~ 2x more disk space than necessary. I'm fairly confident that it must be some sort of GPL-ish license, but you're right that I should seek to confirm the details.

> (you have power to merge, right?)

Nope!

Reply:
Oh, I saw `dontBuild = true` but thought maybe that was a hack because everything's actually getting done in `installPhase` (wasn't sure what the `install.sh` script actually does).

Okay, please just confirm you're satisfied with it and I'm happy to click the merge button.

Reply:
All signs point towards the correct license being GPLv2, including the `relocate_sdk.py` script in the root of the tarball. The contents of the distribution are two Linux sysroots (one which contains the cross-compile toolchain, and the other is the arm64 build target), so I really can't imagine this being something other than GPLv2. I'm happy for this to be merged, thanks!